### PR TITLE
Add F1 Score, Confusion Matrix, and general code edits

### PR DESCRIPTION
**Pull request by chanteriam**
## Describe your changes
Added functionality for calculating the F1 score, confusion matrices, and fixed the `__repr__()` function for XGBoost.

## Checklist before requesting a review

- [x] The code runs successfully.

```
(supreme-court-predictions-py3.11) shaymilner@MacBook-Air-4 supreme-court-ml-predictions % make format
isort "supreme_court_predictions"/ test/
Skipped 1 files
black "supreme_court_predictions"/ test/ *.ipynb
All done! ✨ 🍰 ✨
25 files left unchanged.
(supreme-court-predictions-py3.11) shaymilner@MacBook-Air-4 supreme-court-ml-predictions % make lint
ruff "supreme_court_predictions"/ test/
(supreme-court-predictions-py3.11) shaymilner@MacBook-Air-4 supreme-court-ml-predictions % make test
pytest -vs test/
========================================= test session starts ==========================================
platform darwin -- Python 3.11.1, pytest-7.3.1, pluggy-1.0.0 -- /Users/shaymilner/Library/Caches/pypoetry/virtualenvs/supreme-court-predictions-sGlGetDh-py3.11/bin/python
cachedir: .pytest_cache
rootdir: /Users/shaymilner/Documents/Harris/Spring23/Machine_Learning/final_project/supreme-court-ml-predictions
plugins: anyio-3.6.2
collected 1 item                                                                                       

test/test_util.py::test_get_full_pathway PASSED

========================================== 1 passed in 0.00s ===========================================

```
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: supreme_court_predictions/statistics/descriptives.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import numpy as np
 import pandas as pd
 
-from supreme_court_predictions.util.contants import ENCODING_UTF_8
+from supreme_court_predictions.util.constants import ENCODING_UTF_8
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/descriptives.py:8]: Don't tell anyone I can't spell. 😭😭

File: supreme_court_predictions/models/model.py [hunk 2]
```diff
@@ -21,18 +23,25 @@ def create(self, df):
...
-        :return: Float of some accuracy measurement.
+        :return: Float of accuracy.
+        :return: Float of F1 score.
+        :return: Confusion matrix for model
         """
-        _, y_test, y_pred = self.create(df)
+        model, y_test, y_pred = self.create(df)
 
-        return accuracy_measure(y_true=y_test, y_pred=y_pred)
+        return (
```
Comment (michplunkett) [inline supreme_court_predictions/models/model.py:39]: Noiiiice.

**Review by michplunkett — COMMENTED** [review]
I'll give it one more look over in about 30! Great work so far!

File: supreme_court_predictions/models/model.py [hunk 3]
```diff
@@ -21,18 +23,25 @@ def create(self, df):
...
+            model,
+            accuracy_score(y_true=y_test, y_pred=y_pred),
+            f1_score(y_true=y_test, y_pred=y_pred),
+            confusion_matrix(y_true=y_test, y_pred=y_pred),
```
Comment (michplunkett) [inline supreme_court_predictions/models/model.py:43]: Are these three functions the same for all models?

File: supreme_court_predictions/models/model.py [hunk 4]
```diff
@@ -56,7 +65,11 @@ def __repr__(self):
         """
 
     def print_results(
-        self, model_name="", accuracy_score=None, dataframe_name=None
+        self,
+        model_name="",
+        accuracy_score=None,
+        f1_score=None,
+        dataframe_name=None,
```
Comment (michplunkett) [inline supreme_court_predictions/models/model.py:72]: 🔬 Alpha-sort these bbs if you can.

File: supreme_court_predictions/models/xg_boost.py [hunk 5]
```diff
@@ -147,16 +153,35 @@ def __repr__(self):
...
 
-        s += "ACCURACIES: "
-        for name, acc in zip(self.dataframe_names, self.accuracies):
-            s += f"\n\t{name}: {str(acc)}"
+        return_str += "F1 SCORES: "
+        for name, f1 in self.f1.items():
+            return_str += f"\n\t{name}: {str(f1)}"
 
-        return s
+        return return_str
```
Comment (michplunkett) [inline supreme_court_predictions/models/xg_boost.py:187]: This feels much easier to read, good change.

**Review by michplunkett — APPROVED** [review]
Looks good! Great work, @chanteriam!

**Review by FedericoDM — APPROVED** [review]
:))

File: supreme_court_predictions/models/model.py (hunk 3 above)
Comment (chanteriam) [reply supreme_court_predictions/models/model.py:43]: Yup, they’re general sklearn functions that aren’t model dependent.

**Pull request merged**